### PR TITLE
feat(rds): use user-defined security group for db user rotation function

### LINKED ISSUE #23086: rds: use user-defined security group for db user rotation function
### Describe the feature

When using `addRotationSingleUser` or `addRotationMultiUser` from rds module, the security group of the rotation Lambda function is always automatically created.

Therefore it is difficult for Secrets Manager endpoint to restrict access only to the security group of the rotation Lambda function. You have to allow all traffic to the Secrets Manager endpoint from within the VPC.

### Use Case

We will be able to limit access to the Secrets Manager endpoint from a single security group.

### Proposed Solution

The props of `secretsmanager.SecretRotation` class already has a `securityGroup` property. Just add `securityGroup` property to the options of `addRotationSingleUser` and `addRotationMultiUser` methods and pass it to `secretsmanager.SecretRotation`.

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.50.0

### Environment details (OS name and version, etc.)

macOS Ventura 13.0

### PR DESCRIPTION
**Pull request by clueleaf**
closes #23086 

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by clueleaf** [comment]
Integration test is not added because I think the modification is tested by unit tests.

File: packages/@aws-cdk/aws-rds/README.md [hunk 1]
```diff
@@ -340,6 +340,7 @@ declare const instance: rds.DatabaseInstance;
 instance.addRotationSingleUser({
   automaticallyAfter: cdk.Duration.days(7), // defaults to 30 days
   excludeCharacters: '!@#$%^&*', // defaults to the set " %+~`#$&*()|[]{}:;<>?!'/@\"\\"
+  securityGroup: mySecurityGroup, // optional, defaults to an auto-created security group
```
Comment (mrgrain) [inline packages/@aws-cdk/aws-rds/README.md:343]: please use `declare` to explain what `mySecurityGroup` is
(like `instance` above this)

**Review by mrgrain — DISMISSED** [review]
This looks good.

Please also add or extend an integration test. E.g. `packages/@aws-cdk/aws-rds/test/integ.cluster-rotation.lit.ts` (although note this is an old style tests and it will need converting to the latest syntax)

https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md#integration-tests
https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md

File: packages/@aws-cdk/aws-rds/README.md (hunk 1 above)
Comment (clueleaf) [reply packages/@aws-cdk/aws-rds/README.md]: Thank you for the comment. Fixed!

**Comment by mrgrain** [comment]
@mergifyio update

**Comment by mrgrain** [comment]
@clueleaf Something up with the build here. Can you check how that's related to your change?

**Comment by clueleaf** [comment]
@mrgrain 
Don't fully understand but it seems like `IntegTest` declaration was unnecessary?

**Comment by mrgrain** [comment]
> Don't fully understand but it seems like IntegTest declaration was unnecessary?

@clueleaf I think I know why. The `.lit.ts` files are also used as examples in the docs. Adding `IntegTest` messed up the examples parser. The proper fix would be to:

- Use `IntegTest`
- Remove the `.lit` part
- Replace the usage of the example with an dedicated inline example

or

Create a new testfile without `.lit`.

But this is fine for now. Thanks for bearing with me!

**Review by mrgrain — APPROVED** [review]

**Pull request merged**